lib/Driver/ToolChains.cpp

```diff
@@ -1286,6 +1300,24 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
   if (!Linker.empty()) {
     Arguments.push_back(context.Args.MakeArgString("-fuse-ld=" + Linker));
   }
+  
+  // Configure the toolchain
+  // By default, use the system clang++ to link.
+  const char * Clang = "clang++";
```

Comment:
Clang behaves differently when invoked as `cc`, and I don't know if `cc` (or `c++`) is formally spec'd to do anything sensible when just given object files. We also don't really expect it to be `g++` either—we want to be able to add Clang-specific options if we need to.